docs/proposals/kubectl-apply-purge-from-namespace.md

```diff
@@ -0,0 +1,103 @@
+# Enable purging resources in `kubectl apply`
+
+## Motivation
+
+See [#19805](https://github.com/kubernetes/kubernetes/issues/19805).
+
+Currently `kubectl apply` will create any new resources that exist in the resource file(s) it is supplied with, but it has no way of deleting anything that has been removed from such files.
+
+This makes it difficult to use `kubectl apply` to update reality to match the declarative definitions of resources when that definition evolves over time, since changes can include deletions.
```

Comment:
We have refereed to this process as "configuration reconciliation" in the past. https://github.com/kubernetes/kubernetes/issues/1702